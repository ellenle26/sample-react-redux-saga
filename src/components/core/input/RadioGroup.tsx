import { FC } from "react";


const RadioGroup: FC<RadioGroupProps> = ( props: RadioGroupProps ) => {
    const required = props.required === true;
    const requiredMargin = required ? "mr-2" : "";

    return (
        <div className={`${props.className}`}>
            {(props.title || props.required) && (
                <div className={"mb-1 flex items-center text-sm font-bold"}>
                    {props.title && (
                        <p className={`${requiredMargin}`}>{props.title}</p>
                    )}
                </div>
            )}
            <form id="radio_group" className={props.horizontalDisplay ? "flex-row" : "flex-col"}>
                {
                    Array.isArray(props.options) && props.options.length > 0 && props.options.map((item, index) => (
                        <label key={index} className={"cursor-pointer leading-loose flex items-center mb-1"}>
                            <input
                                id="radio"
                                className={"hidden"}
                                type="checkbox"
                                name={item}
                                checked={props.value === item}
                                onChange={(e) => props.onChange && props.onChange(e.target.name)}
                            />
                            <img
                                className={"inline-block mr-3"}
                                src={
                                    props.value === item
                                        ? "/icons/radio_checked.svg"
                                        : "/icons/radio_unchecked.svg"
                                }
                                alt="icon"
                            />
                            <p className={"text-sm text-gray-800"}>{item}</p>
                        </label>
                    ))
                }
            </form>
        </div>
    )
}

interface RadioGroupProps {
    title?: string;
    required?: boolean;
    horizontalDisplay?: boolean;
    options: string[];
    value: string;
    className?: string;
    onChange?: (value: any) => void;
}

export default RadioGroup;