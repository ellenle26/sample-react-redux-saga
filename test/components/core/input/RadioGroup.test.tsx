/**
 * @jest-environment jsdom
*/

import RadioGroup from "@components/core/input/RadioGroup";
import { fireEvent, render } from "@testing-library/react";
import React from "react";

test('screenshot testing', () => {
    const renderResult = render(<RadioGroup options={["a", "b", "c", "d"]} value="a"/>);
    expect(renderResult.asFragment()).toMatchSnapshot()
})

test('render items testing', () => {
    const renderResult = render(<RadioGroup options={["a", "b", "c", "d"]} value="a" title="test"/>);

    expect(renderResult.container.children[0].children.length).toEqual(2);
    expect(renderResult.container.children[0].children[1].children.length).toEqual(4);
})

test('onchange testing', () => {
    const handleChange = jest.fn();
    const renderResult = render(<RadioGroup options={["a", "b", "c", "d"]} value="a" title="test" onChange={val => handleChange(val)}/>);
    const selectedRadio = renderResult.getByLabelText('c');
    const unselectedRadio = renderResult.getByLabelText('a');
    fireEvent.click(selectedRadio);

    expect(unselectedRadio.ariaChecked).toBeFalsy();
})

test('render items 2 testing', () => {
    const renderResult = render(<RadioGroup options={["a", "b", "c", "d"]} value="a" title="test" required horizontalDisplay />);

    expect(renderResult.container.children[0].children.length).toEqual(2);
    expect(renderResult.container.children[0].children[1].children.length).toEqual(4);
})