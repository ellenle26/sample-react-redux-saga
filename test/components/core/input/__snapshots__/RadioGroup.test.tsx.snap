// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screenshot testing 1`] = `
<DocumentFragment>
  <div
    class="undefined"
  >
    <form
      class="flex-col"
      id="radio_group"
    >
      <label
        class="cursor-pointer leading-loose flex items-center mb-1"
      >
        <input
          checked=""
          class="hidden"
          id="radio"
          name="a"
          type="checkbox"
        />
        <img
          alt="icon"
          class="inline-block mr-3"
          src="/icons/radio_checked.svg"
        />
        <p
          class="text-sm text-gray-800"
        >
          a
        </p>
      </label>
      <label
        class="cursor-pointer leading-loose flex items-center mb-1"
      >
        <input
          class="hidden"
          id="radio"
          name="b"
          type="checkbox"
        />
        <img
          alt="icon"
          class="inline-block mr-3"
          src="/icons/radio_unchecked.svg"
        />
        <p
          class="text-sm text-gray-800"
        >
          b
        </p>
      </label>
      <label
        class="cursor-pointer leading-loose flex items-center mb-1"
      >
        <input
          class="hidden"
          id="radio"
          name="c"
          type="checkbox"
        />
        <img
          alt="icon"
          class="inline-block mr-3"
          src="/icons/radio_unchecked.svg"
        />
        <p
          class="text-sm text-gray-800"
        >
          c
        </p>
      </label>
      <label
        class="cursor-pointer leading-loose flex items-center mb-1"
      >
        <input
          class="hidden"
          id="radio"
          name="d"
          type="checkbox"
        />
        <img
          alt="icon"
          class="inline-block mr-3"
          src="/icons/radio_unchecked.svg"
        />
        <p
          class="text-sm text-gray-800"
        >
          d
        </p>
      </label>
    </form>
  </div>
</DocumentFragment>
`;
